scripts/ci/lib.sh

```diff
@@ -551,15 +556,42 @@ image_prefetcher_start() {
     # pull secret
     NAMESPACE=$ns make -C operator stackrox-image-pull-secret
 
-    # apply daemonset and configmap
+    # apply configmap, daemonset etc
     retry 5 true kubectl apply --namespace=$ns -f "$manifest"
     rm -f "$image_list" "$manifest"
 }
 
 image_prefetcher_await() {
     local ns="prefetch-images"
     local name="$1"
+    local image_prefetcher_metrics="$2"
     kubectl rollout status daemonset "$name" -n "$ns" --timeout 15m
+    # All images fetched, now retrieve metrics.
+    local attempt=0
+    local service="service/${name}-metrics"
+    while [[ -z $(kubectl -n "${ns}" get "${service}" -o jsonpath="{.status.loadBalancer.ingress}" 2>/dev/null) ]]; do
+        if [ "$attempt" -lt "10" ]; then
+            echo "Waiting for ${service} to obtain endpoint ..."
+            ((attempt++))
+            sleep 10
+        else
+            echo "Timeout waiting for ${service} to obtain endpoint!"
+            exit 1
+        fi
+    done
+    local endpoint
+    endpoint="$(kubectl -n "${ns}" get "${service}" -o json | jq -r '.status.loadBalancer.ingress[] | .ip')"
+    local fetcher_metrics
+    fetcher_metrics="$(mktemp --suffix=.csv)"
+    # see the stackrox_image_prefetches table definition for the order of columns
+    "${image_prefetcher_metrics}" "${endpoint}:8443" \
```

Comment:
Probably I miss something but why do we need a tool to download metrics? Why not just curl the endpoint and add a columns with metadata?